src/ui/map.js

```diff
@@ -1023,8 +1028,14 @@ class Map extends Camera {
      * @example
      * const projection = map.getProjection();
      */
-    getProjection() {
-        return this.transform.getProjection();
+    getProjection(): ProjectionSpecification {
+        if (this._explicitProjection) {
+            return this._explicitProjection;
+        }
+        if (this.style && this.style.stylesheet && this.style.stylesheet.projection) {
+            return this.style.stylesheet.projection;
```

Comment:
In the case where we fallback to a stylesheet projection, the projection will be returned in the same format it appears in the stylesheet. i.e. center and parallels can be absent.

This allows us to avoid storing another projection value (`map._runtimeProjection` or `map._implicitProjection`).

The potential downside I see in this is that it could impose a challenge for users when attempting to check if two projections are the same.

One other option to avoid that would be adding a `transform.standardizeProjection(projectionSpec)` function and standardize the projection here before returning.

@karimnaaji @ansis I'd love to hear your thoughts on what the right approach is!

Reply:
>The potential downside I see in this is that it could impose a challenge for users when attempting to check if two projections are the same.
>
>One other option to avoid that would be adding a transform.standardizeProjection(projectionSpec) function and standardize the projection here before returning.

That's a good point, but probably good to follow up after this PR is finalized to make sure it doesn't grow too much out of scope. But the standardization step sounds good to me.

Reply:
Captured in https://github.com/mapbox/mapbox-gl-js/issues/11490 for follow up, marking this comment as resolved.